File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/InferenceEndpoint.js

// Define collection and schema for InferenceEndpoint
export interface InferenceEndpoint {
    name:
	type : string
    endpointUrl:
	type : string
    trafficShare:
	type : Percentage
    ModelVersion:
	type : Schema.Types.ObjectId
    Workspace:
	type : Schema.Types.ObjectId
    Predictions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Prediction' }]
    Mode:
 	type : String
#
    collection: 'inferenceEndpoints'
}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/inferenceendpoint.rb
class InferenceEndpoint < ApplicationRecord
  enum Mode: [:Batch, :RealTime]


  composed_of :percentage,
    class_name: "Percentage",
    mapping: [
      %w[percentage_value value]
    ]

  has_many :ModelVersion, class_name: 'ModelVersion'
  has_many :Workspace, class_name: 'AnalyticsWorkspace'
  has_many :Predictions, class_name: 'Prediction'

end
